From d55153874daaa901540076822187b48b576eb316 Mon Sep 17 00:00:00 2001
From: Sebanc <sebanc.xda@gmail.com>
Date: Wed, 9 Oct 2019 21:49:50 +0200
Subject: [PATCH] Core Microsoft Surface devices: Fix wifi (2/4)

Allowing powersave in cfg80211 module breaks mwifiex module so disable it.
---
 net/wireless/nl80211.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/net/wireless/nl80211.c b/net/wireless/nl80211.c
index 8e2f03a..4c8f617 100644
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -10149,9 +10152,8 @@ static int nl80211_set_power_save(struct sk_buff *skb, struct genl_info *info)
 
 	state = (ps_state == NL80211_PS_ENABLED) ? true : false;
 
-	if (state == wdev->ps)
-		return 0;
-
+	state = false;
+
 	err = rdev_set_power_mgmt(rdev, dev, state, wdev->ps_timeout);
 	if (!err)
 		wdev->ps = state;
-- 
2.20.1

